railgun/internal/ctrlutils/utils.go

```diff
@@ -141,3 +146,31 @@ func CRDExists(client client.Client, gvr schema.GroupVersionResource) bool {
 	}
 	return true
 }
+
+// retrieve Kong Admin API URL from configured name/namespace service
+func RetrieveKongAdminAPIURL(ctx context.Context, KongAdminAPI string, kubeCfg *rest.Config) (string, error) {
+	namespace, name, err := util.ParseNameNS(KongAdminAPI)
+	if err != nil {
+		return "", fmt.Errorf("failed to pars kong admin api namespace and name. err %v", err)
+	}
+	CoreClient, _ := clientset.NewForConfig(kubeCfg)
```

Comment:
There are two problems we need to solve here:

1. we're throwing away the error, which causes fragility which could result in runtime `panics`
2. we don't need to create a new client each time this message is called if the `*kube.Config` is static, we can simply send in the client itself to reduce calls and memory utilization.

My suggestion is to employ number 2 above, and make sure that we check the error.

Reply:
agree that we could not swallow error silently.